xwiki-pro-macros-api/src/main/java/com/xwiki/macros/showhideif/macro/MacroParameter.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.macros.showhideif.macro;
+
+import org.xwiki.properties.annotation.PropertyDescription;
+
+import com.xwiki.macros.internal.grouplist.GroupReferenceList;
+import com.xwiki.macros.internal.userlist.UserReferenceList;
+
+/**
+ * Parameters for {@link com.xwiki.macros.showhideif.macro.HideIfMacro} and
+ * {@link com.xwiki.macros.showhideif.macro.ShowIfMacro} Macros.
+ *
+ * @version $Id: $
+ */
+public class MacroParameter
+{
+    private MacroParamMatchUsing matchUsing = MacroParamMatchUsing.ANY;
+
+    private MacroParamAuthenticationType authenticationType = MacroParamAuthenticationType.NONE;
+
+    private UserReferenceList users;
+
+    private GroupReferenceList groups;
+
+    /**
+     * @version $Id: $
+     */
+    public enum MacroParamMatchUsing
```

Comment:
Since its a nested class, I think we can drop the prefix and simply name it `Matcher`. When we access it, it will be something like `ShowHideIfMacroParameters.Matcher.ANY`.